File: Backend/TopSys.TopConWeb.Domain/Entities/Lead/LeadInteracao.cs
using System;

namespace TopSys.TopConWeb.Domain.Entities.Lead
{
    public class LeadInteracao
    {
        public Guid Id { get; set; }
        public int Usina { get; set; }
        public int NumeroLead { get; set; }
        public int AnoLead { get; set; }
        public string Tipo { get; set; }
        public string Descricao { get; set; }
        public DateTime Data { get; set; }
        public TimeSpan Hora { get; set; }
        public DateTime? DataRetorno { get; set; }
        public TimeSpan? HoraRetorno { get; set; }
        public string IdCadastro { get; set; }
    }
}

File: Frontend/src/app/classes/lead/lead-interacao.ts
export enum ELeadInteracaoTipo {
    AtividadeRealizada = 1,
    InteracaoCliente = 2,
    Chamada = 3,
    Todos = 99
}

export const LeadInteracaoTipos = [
    { codigo: ELeadInteracaoTipo.Todos, descricao: 'Todos', sigla: '' },
    { codigo: ELeadInteracaoTipo.AtividadeRealizada, descricao: 'Atividade Realizada', sigla: 'ATIVIDADE' },
    { codigo: ELeadInteracaoTipo.InteracaoCliente, descricao: 'Interação com Cliente', sigla: 'INTERACAO' },
    { codigo: ELeadInteracaoTipo.Chamada, descricao: 'Chamada', sigla: 'CHAMADA' },
];

export class LeadInteracao {
    usina: number = 0;
    numeroLead: number = 0;
    anoLead: number = 0;
    tipo: string = '';
    descricao: string = '';
    data: Date = new Date();
    hora: string = new Date().getHours().toString().padStart(2, '0') + ':' + new Date().getMinutes().toString().padStart(2, '0') + ':00';
    horaString: string = new Date().getHours().toString().padStart(2, '0') + new Date().getMinutes().toString().padStart(2, '0');
    dataRetorno: Date = null;
    horaRetorno: string = '';
    horaRetornoString: string = '';
    idCadastro: string = '';
}